org-roam.el

```diff
@@ -519,1159 +301,580 @@ Use external shell commands if defined in `org-roam-list-files-commands'."
-                       (eq org-roam-link-use-custom-faces 'everywhere))))
-      (cond ((and custom
-                  (not (file-remote-p path)) ;; Prevent lockups opening Tramp links
-                  (not (file-exists-p path)))
-             'org-roam-link-invalid)
-            ((and (org-roam--in-buffer-p)
-                  (org-roam--backlink-to-current-p))
-             'org-roam-link-current)
-            ((and custom
-                  (org-roam--org-roam-file-p path))
-             'org-roam-link)
-            (t
-             'org-link)))))
-
-(defun org-roam--id-link-face (id)
-  "Conditional face for id links.
-Applies `org-roam-link-current' if ID corresponds to the
-currently opened Org-roam file in the backlink buffer, or
-`org-roam-link-face' if ID corresponds to any other Org-roam
-file."
-  (save-match-data
-    (let* ((in-note (-> (buffer-file-name (buffer-base-buffer))
-                        (org-roam--org-roam-file-p)))
-           (custom (or (and in-note org-roam-link-use-custom-faces)
-                       (eq org-roam-link-use-custom-faces 'everywhere))))
-      (cond ((and (org-roam--in-buffer-p)
-                  (org-roam--backlink-to-current-p))
-             'org-roam-link-current)
-            ((and custom
-                  (org-roam-id-get-file id t))
-             'org-roam-link)
-            ((and custom
-                  (not (org-roam-id-get-file id)))
-             'org-roam-link-invalid)
-            (t
-             'org-link)))))
-
-;;;; Hooks and Advices
-(defcustom org-roam-file-setup-hook nil
-  "Hook that is run on setting up an Org-roam file."
-  :group 'org-roam
-  :type 'hook)
-
-(defun org-roam--find-file-hook-function ()
-  "Called by `find-file-hook' when mode symbol `org-roam-mode' is on."
-  (when (org-roam--org-roam-file-p)
-    (setq org-roam-last-window (get-buffer-window))
-    (run-hooks 'org-roam-file-setup-hook) ; Run user hooks
-    (org-roam--setup-title-auto-update)
-    (add-hook 'post-command-hook #'org-roam-buffer--update-maybe nil t)
-    (add-hook 'before-save-hook #'org-roam-link--replace-link-on-save nil t)
-    (add-hook 'after-save-hook #'org-roam-db-update nil t)
-    (dolist (fn org-roam-completion-functions)
-      (add-hook 'completion-at-point-functions fn nil t))
-    (org-roam-buffer--update-maybe :redisplay t)))
+  "Return all titles and aliases in the Org-roam database."
+  (let* ((titles (mapcar #'car (org-roam-db-query [:select title :from nodes])))
+         (aliases (mapcar #'car (org-roam-db-query [:select alias :from aliases])))
+         (completions (append titles aliases)))
+    completions))
+
+;;; Org-roam setup and teardown
+(defvar org-roam-find-file-hook nil
+  "Hook run when an Org-roam file is visited.")
+
+(defun org-roam-setup ()
+  "Setup Org-roam."
+  (interactive)
+  (unless (or (and (bound-and-true-p emacsql-sqlite3-executable)
+                   (file-executable-p emacsql-sqlite3-executable))
```

Comment:
Looks like these may be intended to be `emacsql-sqlite-executable` as the ones here aren't defined in emacsql.